application/main.py

```diff
@@ -794,6 +794,31 @@ def get_model(fn):
         return response
     else:
         return send_file(path)
+    
+@application.route('/preprocess_bsdd/<id>', methods=['GET'])
+@login_required
+def preprocess_bsdd(user_data, id):
+    with database.Session() as session:
+        model = session.query(database.model).filter(
+            database.model.code == id).first()
+        bsdd_task = [task for task in model.tasks if task.task_type == "bsdd_validation_task"][0]
```

Comment:
We need to validate whether the currently logged-in user is the owner of the file.

```suggestion
        if model.user_id != user_data["sub"]:
            abort(403)
        bsdd_task = [task for task in model.tasks if task.task_type == "bsdd_validation_task"][0]
```